nixos/modules/system/activation/top-level.nix

```diff
@@ -127,7 +127,7 @@ let
       configurationName = config.boot.loader.grub.configurationName;
 
       # Needed by switch-to-configuration.
-      perl = "${pkgs.perl}/bin/perl -I${pkgs.perlPackages.FileSlurp}/lib/perl5/site_perl";
+      perl = "${pkgs.perl}/bin/perl -I${pkgs.perlPackages.FileSlurp}/lib/perl5/site_perl -I${pkgs.perlPackages.NetDBus}/lib/perl5/site_perl -I${pkgs.perlPackages.XMLParser}/lib/perl5/site_perl -I${pkgs.perlPackages.XMLTwig}/lib/perl5/site_perl";
```

Comment:
Can be cleaned up to `${concatStringsSep " " (map (lib: "-I${lib}/${pkgs.perl.libPrefix}") (with pkgs.perlPackages; [ FileSlurp NetDBus XMLParser XMLTwig ]))}`. Is not there an existing function (like `python.withPackages`) for that?

Reply:
I don't know of one, and digging around naively didn't find me it. I could most definitely write one, though.